### ISSUE TITLE
Auto add self dependency to npm

### ISSUE THREAD

**Issue by tienchai-ample** [fabric8-analytics/fabric8-analytics-vscode-extension#732]
Type: <b>Bug</b>

When I open a package.json file (for npm), it seems that these 2 things happen:
1. a self dependency is added into the package.json file
2. package-lock.json is generated. 

This is as-if `npm install .` was called every time I open a package.json file, on the package.json directory. 

Extension version: 0.9.5
VS Code version: Code 1.92.0 (b1c0a14de1414fcdaa400695b4db1c0799bc3124, 2024-07-31T23:26:45.634Z)
OS version: Windows_NT x64 10.0.22631
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|13th Gen Intel(R) Core(TM) i7-1360P (16 x 2611)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|31.68GB (18.59GB free)|
|Process Argv|--crash-reporter-id 1fede22c-cab8-45ed-8080-59ceef7a5290|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscorecescf:30445987
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
a9j8j154:30646983
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonnoceb:30805159
asynctok:30898717
pythonregdiag2:30936856
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
accentitlementst:30995554
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
da93g388:31013173
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
dwnewjupytercf:31046870
impr_priority:31102340
refactort:31108082
ccplt:31103426
pythonrstrctxt:31103193
wkspc-onlycs-c:31106320
wkspc-ranged-c:31107834

```

</details>

**Comment by Joniras** [comment]
I had to enable/disable every extension by hand to find out it is this extension preventing me basically to build my library.
More details from me:
Everytime i alter/open a package.json and delete the line
```json
    "@libraryname/ngx-table": "file:",
```

it gets added again and prevents me to build. 
I alos get the same behaviour in another project where "webapp": "file:" is added (webapp is the name of the folder the project is in).

**Comment by pmeske** [comment]
I can confirm this Issue with Version v0.9.5

**Comment by scoorpion1008** [comment]
It seems like you’re encountering an issue where opening a package.json file in Visual Studio Code triggers behavior similar to running npm install ., resulting in a self-dependency being added and a package-lock.json file being generated. This is not the expected behavior when simply opening a file. a few steps you can take to troubleshoot and potentially resolve the issue:

Check for Extension Conflicts: Sometimes, extensions can cause unexpected behavior. [Check if you have any extensions related to npm or package management that might be causing this issue](https://marketplace.visualstudio.com/items?itemName=ldd-vs-code.better-package-json)[1](https://marketplace.visualstudio.com/items?itemName=ldd-vs-code.better-package-json).
Update VS Code and Extensions: Ensure that your VS Code and all extensions are up to date. An outdated extension or VS Code version might have known bugs that have been fixed in later releases.
Review Extension Settings: Look into the settings of your installed extensions to see if there’s an option that might be causing this behavior. Some extensions have features that automatically run npm commands upon certain triggers.
Disable Extensions: Try disabling all extensions and then re-enable them one by one to identify if a specific extension is causing the problem.
Check VS Code Settings: Review your VS Code settings (both user and workspace settings) for any configurations related to npm or scripts that might be running on file open.
Consult VS Code Documentation: The VS Code documentation and community forums can be a valuable resource for troubleshooting issues. [You might find others who have experienced similar problems and shared solutions](https://marketplace.visualstudio.com/items?itemName=ldd-vs-code.better-package-json)[2](https://code.visualstudio.com/api/working-with-extensions/bundling-extension).
Report the Issue: If none of the above steps help, consider reporting the issue to the VS Code GitHub repository with detailed information about your setup and the steps to reproduce the bug.
Remember to back up your package.json and package-lock.json files before making changes or updates, just in case you need to revert to a previous state. Hopefully, these steps will help you resolve the issue and prevent the unwanted behavior from occurring.

**Comment by gian1200** [comment]
Is that a bot comment?

Regarding the issue, it happens on last version only (0.9.5). As a workaround one can force `0.9.4` or disable the dependency and rollback the changes (I did the 2nd one) until a new version with a fix is released.

**Comment by scoorpion1008** [comment]
Iam Mexican....lol was just trying to be helpful by providing precise/concise alternatives for troubleshooting the possible issues 
and regarding the package. Json it’s good to hear that you’ve identified the issue as being specific to version 0.9.5 of the extension and that you’ve found a workaround forcing 0.9.4 disabling the dependency and rolling back the changes a temporary solution until the extension’s maintainers releases a new version with a fix. i would hit them up with the issue or use code pilot in vscode for a robust suggestion

**Comment by yostane** [comment]
I also have the above described issue with this additional effect:

- Autoformat of package.json on file open
- Autoadd of self dependency in package-lock.json
- AFAIK No logs anywhere in the output windows

By switching to 0.9.4 the issue disappeared, but I get a warning notification `Command failed: npm i --package-lock-only --prefix ...`

**Comment by GuiRitter** [comment]
This is specially annoying after I moved to `yarn`, as it keeps downloading the dreaded `node_modules` which I finally managed to get myself rid of...

**Comment by Astralzz** [comment]
I had the same problem and couldn't compile my React Native app either.

The issue was caused by the [Red Hat Dependency Analytics](https://marketplace.visualstudio.com/items?itemName=redhat.fabric8-analytics) extension. I uninstalled and disabled it, and the problem was resolved

**Comment by yostane** [comment]
> I had the same problem and couldn't compile my React Native app either.
> 
> The issue was caused by the [Red Hat Dependency Analytics](https://marketplace.visualstudio.com/items?itemName=redhat.fabric8-analytics) extension. I uninstalled and disabled it, and the problem was resolved

Just in case,. if you still want to use the extension, you can try to install version v0.9.4

**Comment by carlosthe19916** [comment]
Hi everyone. I am trying to understand where the problem is, and if possible try to fix it, but I was not able to reproduce the issue. With so many people reporting this I guess I am missing something but I don't know what. Could you help me please?



- I installed the extension v0.9.5
- Open a Javascript project
- Open the Dependency Analytics Report
- I did not see a self dependency being added to my package.json file as reported in this issue.

## General points

- I investigated and the extension currently supports only NPM and not YARN. Could you confirm you have this issue using NPM?
- I verified that indeed the package-lock.json is generated but only if you did not have a package-lock.json file previously, if you had one previously then your file remains the same and untouched. Could you anybody confirm you see this issue because you did not have a package-lock.json previously? All npm projects need/should/must have a package-lock.json together with the package.json it is a good practive/standard.

> The video was recorded using this UI project https://github.com/project-openubl/searchpe/tree/master/application/src/main/webapp

Can you guys help me to reproduce the issue? If you could share a project based on npm that I can use for testing that would be great too.

**Comment by carlosthe19916** [comment]
Thanks @yostane I will give it a try thanks a lot... I will be a bit busy with https://github.com/RHEcosystemAppEng/exhort-javascript-api/pull/151 and then try to jump to this one.

**Comment by ruromero** [comment]
Correct, there are 2 issues described in the bug. The package.json generation is already fixed https://github.com/trustification/exhort-javascript-api/pull/151 but not yet released and the self-dependency issue seems to be related to an incorrect use of the path when doing npm install.

As npm install will no longer be executed if the package-lock.json doesn't exist I can assume this won't happen either but we will try to validate it before releasing a new version of the plugin.
